internal/scheduler/constraints/constraints.go

```diff
@@ -0,0 +1,167 @@
+	//
+	// Cumulative resource usage at priority x includes resources allocated to jobs of priorityClassPriority x or lower.
+	MaximumCumulativeResourceFractionPerQueue map[string]float64
+}
+
+func SchedulingConstraintsFromSchedulingConfig(
+	minimumJobSize schedulerobjects.ResourceList,
+	config configuration.SchedulingConfig,
+) SchedulingConstraints {
+	priorityClassSchedulingConstraintsByPriorityClassName := make(map[string]PriorityClassSchedulingConstraints, len(config.Preemption.PriorityClasses))
+	for name, priorityClass := range config.Preemption.PriorityClasses {
+		priorityClassSchedulingConstraintsByPriorityClassName[name] = PriorityClassSchedulingConstraints{
+			PriorityClassName:                         name,
+			PriorityClassPriority:                     priorityClass.Priority,
+			MaximumCumulativeResourceFractionPerQueue: priorityClass.MaximalResourceFractionPerQueue,
+		}
+	}
+	return SchedulingConstraints{
+		MaximumJobsToSchedule:           config.MaximumJobsToSchedule,
+		MaximumGangsToSchedule:          config.MaximumGangsToSchedule,
+		MaxLookbackPerQueue:             config.QueueLeaseBatchSize,
+		MinimumJobSize:                  minimumJobSize,
+		MaximalResourceFractionPerQueue: config.MaximalResourceFractionPerQueue,
+		PriorityClassSchedulingConstraintsByPriorityClassName: priorityClassSchedulingConstraintsByPriorityClassName,
+		MaximalResourceFractionToSchedulePerQueue:             config.MaximalResourceFractionToSchedulePerQueue,
+		MaximalResourceFractionToSchedule:                     config.MaximalClusterFractionToSchedule,
+	}
+}
+
+func (constraints *SchedulingConstraints) CheckGlobalConstraints(ctx context.Context, sctx *schedulercontext.SchedulingContext) (bool, string, error) {
+	if constraints.MaximumJobsToSchedule != 0 && sctx.NumScheduledJobs > int(constraints.MaximumJobsToSchedule) {
+		return false, UnschedulableReasonMaximumNumberOfJobsScheduled, nil
+	}
+	if constraints.MaximumGangsToSchedule != 0 && sctx.NumScheduledGangs > int(constraints.MaximumGangsToSchedule) {
+		return false, UnschedulableReasonMaximumNumberOfGangsScheduled, nil
+	}
+
+	// MaximalResourceFractionToSchedule check.
+	totalScheduledResources := sctx.ScheduledResourcesByPriority.AggregateByResource()
+	if exceeded, reason := exceedsResourceLimits(
+		ctx,
+		totalScheduledResources,
+		sctx.TotalResources,
+		constraints.MaximalResourceFractionToSchedule,
+	); exceeded {
+		unschedulableReason := reason + " (overall per scheduling round limit)"
+		return false, unschedulableReason, nil
+	}
+
+	return true, "", nil
+}
+
+func (constraints *SchedulingConstraints) CheckPerQueueAndPriorityClassConstraints(
+	ctx context.Context,
+	sctx *schedulercontext.SchedulingContext,
+	queue string,
+	priorityClassName string,
+) (bool, string, error) {
+	// We assume that all jobs in a gang have the same priority class (which we enforce at job submission).
```

Comment:
It's not clear to me why this comment is here as the check immediately following this doesn't seem to care about gangs or priority.  Should it be moved to line 114?  (Or even removed completely)

Reply:
Fixed